### PR TITLE
feat: add chat sse support

### PR DESCRIPTION
**Pull request by Abirdcfly**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it
need merge #325 first.

```bash
curl -XPOST http://127.0.0.1:8081/chat --data '{"query":"旷工最小计算单位为多少天？","response_mode":"streaming","conversion_id":"","app_name":"base-chat-with-knowledgebase", "app_namespace":"arcadia"}'

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"旷","created_at":"2023-12-08T16:14:53.884221628+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"工","created_at":"2023-12-08T16:14:53.884294724+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"最小","created_at":"2023-12-08T16:14:53.884316254+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"计算","created_at":"2023-12-08T16:14:53.884363911+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"单位","created_at":"2023-12-08T16:14:53.884377425+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"为","created_at":"2023-12-08T16:14:53.88438656+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"多少","created_at":"2023-12-08T16:14:53.884436339+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"天","created_at":"2023-12-08T16:14:53.884488861+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"？","created_at":"2023-12-08T16:14:53.884505665+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"\n\n旷","created_at":"2023-12-08T16:14:53.88451932+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"工","created_at":"2023-12-08T16:14:53.884555792+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"最小","created_at":"2023-12-08T16:14:53.884584971+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"计算","created_at":"2023-12-08T16:14:53.884604758+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"单位","created_at":"2023-12-08T16:14:53.884668852+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"为","created_at":"2023-12-08T16:14:53.884695446+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" 0","created_at":"2023-12-08T16:14:53.884711674+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":".","created_at":"2023-12-08T16:14:53.884724925+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"5","created_at":"2023-12-08T16:14:53.884793234+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" 天","created_at":"2023-12-08T16:14:53.884816072+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"。","created_at":"2023-12-08T16:14:53.884834661+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"\n\nEx","created_at":"2023-12-08T16:14:53.884851153+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"plan","created_at":"2023-12-08T16:14:53.884867425+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"ation","created_at":"2023-12-08T16:14:53.884885341+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":":","created_at":"2023-12-08T16:14:53.884902908+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" The","created_at":"2023-12-08T16:14:53.884917726+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" given","created_at":"2023-12-08T16:14:53.884975377+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" context","created_at":"2023-12-08T16:14:53.885320007+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" states","created_at":"2023-12-08T16:14:53.88536734+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" that","created_at":"2023-12-08T16:14:53.88538355+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" 旷","created_at":"2023-12-08T16:14:53.885399006+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"工","created_at":"2023-12-08T16:14:53.88541911+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"最小","created_at":"2023-12-08T16:14:53.885483975+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"计算","created_at":"2023-12-08T16:14:53.885559245+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"单位","created_at":"2023-12-08T16:14:53.885691011+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"为","created_at":"2023-12-08T16:14:53.885740829+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" 0","created_at":"2023-12-08T16:14:53.885762678+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":".","created_at":"2023-12-08T16:14:53.885778956+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"5","created_at":"2023-12-08T16:14:53.885796323+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":" 天","created_at":"2023-12-08T16:14:53.885814004+08:00"}

data:{"conversion_id":"708bf843-4958-4e1d-bf95-79d4190e692f","message_id":"","message":"。","created_at":"2023-12-08T16:14:53.885831334+08:00"}


```
#### Which issue(s) this PR fixes

Fixes #

### CODE REVIEW

File: pkg/application/chain/stream.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package chain
```
Comment (nkwangleiGIT) [inline pkg/application/chain/stream.go:1]: add license header

File: pkg/application/retriever/knowledgebaseretriever.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	obj, err = cli.Resource(schema.GroupVersionResource{Group: v1alpha1.GroupVersion.Group, Version: v1alpha1.GroupVersion.Version, Resource: "vectorstores"}).
+		Namespace(vectorStoreReq.GetNamespace()).Get(ctx, vectorStoreReq.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), vectorStore)
+	if err != nil {
+		return nil, err
+	}
+	url := "http://localhost:8888"
```
Comment (nkwangleiGIT) [inline pkg/application/retriever/knowledgebaseretriever.go:110]: still use localhost?

File: api/app-node/chain/v1alpha1/retrievalqachain_types.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	node "github.com/kubeagi/arcadia/api/app-node"
+	"github.com/kubeagi/arcadia/api/base/v1alpha1"
+)
+
+// RetrievalQAChainSpec defines the desired state of RetrievalQAChain
+type RetrievalQAChainSpec struct {
```
[the remark this answers is not in the document]
Comment (Abirdcfly) [reply api/app-node/chain/v1alpha1/retrievalqachain_types.go:27]: Update #325

File: pkg/llms/zhipuai/sse_client.go [hunk 4]
```diff
@@ -65,7 +65,7 @@ func Stream(apiURL, token string, params ModelParams, timeout time.Duration, han
 	}
 	defer resp.Body.Close()
 
-	fmt.Printf("%v", resp)
+	// fmt.Printf("%v", resp)
```
Comment (bjwswang) [inline pkg/llms/zhipuai/sse_client.go:68]: we can remove this line

**Pull request merged**